ts/date-formatter.ts

```diff
@@ -0,0 +1,82 @@
+import { getWeekDates } from "./dateManipulation.js";
+
+export type DateFormatter = ReturnType<typeof createDateFormatter>;
+
+export const createDateFormatter = (locale: string) => {
+	let currentLocale = locale;
+
+	return {
+		getFormattedMonth: (date: Date): string =>
+			date.toLocaleDateString(currentLocale, {
+				month: "long",
+			}),
+		getWeekDayLabels: (date: Date): string[] => {
+			return getWeekDates(date).map((date: Date) => {
+				return date
+					.toLocaleDateString(currentLocale, { weekday: "short" })
+					.toUpperCase();
+			});
+		},
+		getWeekDateLabel: (date: Date): string => {
+			return date
+				.toLocaleDateString(currentLocale, { weekday: "short" })
+				.toUpperCase();
+		},
+		getWeekDates: (date: Date): string[] => {
+			return getWeekDates(date).map((date: Date) => {
```

Comment:
I know this works, but woah this looks really funky 😄 
Maybe import the `getWeekDates` as `_getWeekDates` ? Just for readibility.